### PR TITLE
style: Refactor modal width

### LINKED ISSUE #9242: `modal-width` class to tailwind
same as #9241, replace `modal-width` class with tailwind equivalent so we can remove variable.scss import 

(`w-[unset] lg:w-[25rem]`)

### PR DESCRIPTION
**Pull request by pbkompasz**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring

## Context

- [X] Closes #9242
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [X] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt&usdamount=50&donation=true)

#### Community participation

- [ ] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by preschian — APPROVED** [review]

**Review by roiLeo — APPROVED** [review]

**Comment by prury** [comment]
@pbkompasz please check https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=1458-19091&mode=design&t=sR4VNniVEAbO1Ghi-0 for proper modal width

i understand that on canary it was like 400px but now we have: 
image

@exezbcz 380px, correct?

**Comment by exezbcz** [comment]
> 380px, correct?

yup, it is mobile width

File: components/shared/modals/ModalBody.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="modal-width">
+  <div class="w-[unset] lg:modal-width">
```
Comment (roiLeo) [inline components/shared/modals/ModalBody.vue:2]: use `md:w-[]`

File: components/massmint/modals/modals.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@import '@/assets/styles/abstracts/variables';
```
Comment (roiLeo) [inline components/massmint/modals/modals.scss:1]: remove this file if not used

File: components/shared/modals/ModalBody.vue [hunk 3]
```diff
@@ -114,19 +115,13 @@ watch(
...
 $b-padding: 1.25rem;
 
 .modal-width {
   width: v-bind(modalWidth);
-  max-width: v-bind(modalMaxHeight);
-
-  @include mobile {
-    width: unset;
-  }
+  max-width: v-bind(modalMaxWidth);
```
Comment (roiLeo) [inline components/shared/modals/ModalBody.vue:124]: I don't think we need this anymore, any solution using tailwind?

**Review by roiLeo — CHANGES_REQUESTED** [review]

**Comment by pbkompasz** [comment]
@prury @exezbcz This is what I have now, after merge.
Is this the right width?
Screenshot from 2024-02-08 13-09-24

File: components/shared/modals/ModalBody.vue (hunk 3 above)
Comment (pbkompasz) [reply components/shared/modals/ModalBody.vue]: This?
``` vue
<div :class="`w-[unset] md:v-[${modalWidth}] md:max-w-[${modalMaxWidth}]`">
```

**Comment by roiLeo** [comment]
> This is what I have now, after merge. Is this the right width? 

may [this page](https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=3503-8697&mode=design&t=HqapUqE9GIs42XnT-0) help, should be ~380px

File: components/shared/modals/ModalBody.vue (hunk 3 above)
Comment (roiLeo) [reply components/shared/modals/ModalBody.vue]: yes this is better IMO

File: components/shared/modals/ModalBody.vue [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="modal-width">
+  <div :class="`w-[unset] md:v-[${modalWidth}] md:max-w-[${modalMaxWidth}]`">
```
Comment (roiLeo) [inline components/shared/modals/ModalBody.vue:2]: typo?
```suggestion
  <div :class="`w-[unset] md:w-[${modalWidth}] md:max-w-[${modalMaxWidth}]`">
```

File: components/massmint/modals/modals.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.border-grey {
```
Comment (roiLeo) [inline components/massmint/modals/modals.scss:1]: this file is not used

**Review by roiLeo — COMMENTED** [review]
otherwise lg

note: `NeoModal` & `ModalBody` components need a bit more refactoring to tailwind

**Comment by prury** [comment]
> otherwise lg
> 
> note: `NeoModal` & `ModalBody` components need a bit more refactoring to tailwind

does it means modals will still have different width? 

auto teleport:

**Comment by roiLeo** [comment]
> does it means modals will still have different width?

Yes, some of them will have different width

**Comment by exezbcz** [comment]
@roiLeo that is not expected imo

**Comment by exezbcz** [comment]
any updates @pbkompasz ? 👀

**Comment by pbkompasz** [comment]
@exezbcz I'll finish it tomorrow. Sorry for the delay.

**Comment by prury** [comment]
inconsistencies i could find: 

autoteleport confirm - 350px (400 previously)
delete nfts(in collection) - 450px (400 previously)
list modal - 336px (385 previously)
burn item - 450px (400 previously)
buy now loading - 360px (400 previously)
sign transaction - 450px(was 400 previously)

ideal is around 380px(a bit more or less, np) after tailwind class replacement
again, if this PR is going to introduce a better way to handle modal width and another PR will come to standardize it, let me know

**Comment by prury** [comment]
autoteleport welcome: 

widht PR auto tele

**Comment by pbkompasz** [comment]
Screenshot from 2024-02-27 15-20-07

File: libs/ui/src/components/NeoModal/NeoModal.vue [hunk 6]
```diff
@@ -3,7 +3,11 @@
     <o-modal
       v-model:active="isModalActive"
       class="neo-modal"
-      :class="{ 'append-to-body': appendToBody }"
+      :class="[
+        `max-h-[${maxHeight}] mx-auto`,
+        { 'append-to-body': appendToBody },
+      ]"
+      :style="{ 'max-width': !isMobile ? width : '100%' }"
```
Comment (preschian) [inline libs/ui/src/components/NeoModal/NeoModal.vue:10]: please use tailwind responsive helper instead https://tailwindcss.com/docs/responsive-design

**Review by preschian — CHANGES_REQUESTED** [review]

**Comment by exezbcz** [comment]
@pbkompasz can you please have a look so we can finish this one?

File: components/common/autoTeleport/AutoTeleportWelcomeModal.vue [hunk 7]
```diff
@@ -4,8 +4,9 @@
     :can-cancel="['outside', 'escape']"
     scroll="clip"
     class="z-[1000]"
+    width="40rem"
     @close="onClose">
-    <div class="modal-width">
+    <div class="w-full">
```
Comment (Jarsen136) [inline components/common/autoTeleport/AutoTeleportWelcomeModal.vue:9]: It should not be fixed width

**Pull request closed without merging**